File: src/main/java/me/falsehonesty/asmhelpermod/core/AsmHelperLoadingPlugin.java
package me.falsehonesty.asmhelpermod.core;

import me.falsehonesty.asmhelpermod.AsmHelperMod;
import net.minecraftforge.fml.relauncher.IFMLLoadingPlugin;

import java.util.Map;

public class AsmHelperLoadingPlugin implements IFMLLoadingPlugin {
    @Override
    public String[] getASMTransformerClass() {
        return new String[] {
            "me.falsehonesty.asmhelper.example.TestClassTransformer"
        };
    }

    @Override
    public String getModContainerClass() {
        return AsmHelperMod.class.getName();
    }

    @Override
    public String getSetupClass() {
        return null;
    }

    @Override
    public void injectData(Map<String, Object> data) { }

    @Override
    public String getAccessTransformerClass() {
        return null;
    }
}

File: src/main/kotlin/me/falsehonesty/asmhelper/coremod/AsmHelperLoadingPlugin.kt
package me.falsehonesty.asmhelper.coremod

//#if MC<=11202
import net.minecraftforge.fml.relauncher.IFMLLoadingPlugin

class AsmHelperLoadingPlugin : IFMLLoadingPlugin {
    override fun getASMTransformerClass(): Array<String> {
        return arrayOf("me.falsehonesty.asmhelper.example.TestClassTransformer")
    }

    override fun getModContainerClass(): String? = null

    override fun getSetupClass(): String? = null

    override fun injectData(data: MutableMap<String, Any>?) { }

    override fun getAccessTransformerClass(): String? = null
}
//#endif
